controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,7 +139,42 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 		return response, errors.Errorf("there are %d healthy etcd pods, but %d etcd members", expectedMembers, len(knownMemberIDSet))
 	}
 
-	return response, nil
+	// Check etcd cluster alarms
+	etcdClient, err := w.etcdClientGenerator.forNodes(ctx, controlPlaneNodes.Items)
+	if err != nil {
+		return response, err
+	}
+	defer etcdClient.Close()
+
+	members, err := etcdClient.Members(ctx)
+	if err != nil {
+		return response, err
+	}
+
+	errList := []error{}
+
+	alarmList, err := etcdClient.Alarms(ctx)
+	if len(alarmList) > 0 || err != nil {
+		errList = append(errList, errors.Errorf("etcd cluster has %d alarms", len(alarmList)))
+	}
+
+	healthyMembers := 0
+	for _, m := range members {
```

Comment:
IMO this loop is actually implementing two things:
1. check that all the etcd members have a corresponding node
2. determine which node are healthy

I think we should split the two different for better readability, and possible move 1 up in this func after the for loop which is already implementing other consistency checks.

WRT to 2, my major concern is the fact that we are assessing quorum on the bases of checks that most probably does not correspond to the ones implemented in etcd, so this could lead to both false positive or false negative reports.
If we can't find a way to make this consistent with etcd, I'm leaning toward simplifying this PR and getting only the isEndpointHealthy check added

Reply:
Only left if a member is part of nodes or not check here.